.github/workflows/ci.yml

```diff
@@ -13,38 +11,36 @@ on:
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
   build:
-    # The type of runner that the job will run on
-    runs-on: macos-latest
+    runs-on: macos-13
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-
       # Checks-out the kiwix/apple repository
       - uses: actions/checkout@v4
         with:
           repository: kiwix/apple
           ref: main
           path: apple
     
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      # Checks-out the kiwix/kiwix-custom-apple repository
       - uses: actions/checkout@v4
         with:
           repository: kiwix/kiwix-apple-custom
-          ref: main
+          ref: feature/1-DWDS-build-script
           path: custom
+          
+      # Install the python dependencies
+      - run: python -m pip install pyyaml
 
-      - run: ls -la
-      - run: ls -la custom/
-      - run: ls -la apple/
+      # Install main project dependencies
+      - run: cd apple && brew bundle && cd ..
 
-      # copy build file to the main folder and run from there
+      # copy build file to the main folder to be run from there
       - run: cp custom/build_project.zsh .
```

Comment:
Please create logical blocs of commands like here for example https://github.com/openzim/libzim/blob/main/.github/workflows/ci.yml#L31

Reply:
I have updated this to be more descriptive.